src/main/groovy/life/qbic/datamodel/datasets/OxfordNanoporeMeasurement.groovy

```diff
@@ -0,0 +1,142 @@
+                    folders["fastqpass"] = element as FastQPassFolder
+                    break
+                case FastQFailFolder:
+                    folders["fastqfail"] = element as FastQFailFolder
+                    break
+                case DataFile:
+                    logFiles.add(element as DataFile)
+            }
+        }
+    }
+
+    /**
+     * This method aggregates all fast5 files and fastq files of an Oxford Nanopore
+     * measurement by sample id.
+     *
+     * The resulting data-structure follows this map schema:
+     *
+     * "QBiC sample id":
+     *      "fast5fail": DataFolder
+     *      "fast5pass": DataFolder
+     *      "fastqfail": DataFolder
+     *      "fastqpass": DataFolder
+     * "Other sample id":   // In case of pooled samples
+     *      ...
+     * @return Map A nested map with sample ids and containing data folders
+     */
+    Map<String, Map<String, DataFolder>> getRawDataPerSample(Experiment experiment) {
+        if (pooledMeasurement) {
+            return prepareRawDataFromPooledMeasurement()
+        } else {
+            return prepareRawData(experiment.sampleId)
+        }
+    }
+
+    private Map<String, Map<String, DataFolder>> prepareRawDataFromPooledMeasurement() {
+        final def result = new HashMap()
+        final def pooledSampleIds = folders
+                .get("fast5failed")
+                .getTheChildren()
+                .collect { (it as BarcodedFolder).getSampleId() }
+        pooledSampleIds.each { sampleId ->
+            final def map = [
+                    "fast5fail": (folders.get("fast5failed") as DataFolder).getTheChildren().find {(it as BarcodedFolder).getSampleId()},
+                    "fast5pass": (folders.get("fast5passed") as DataFolder).getTheChildren().find {(it as BarcodedFolder).getSampleId()},
+                    "fastqpass": (folders.get("fastqpass") as DataFolder).getTheChildren().find {(it as BarcodedFolder).getSampleId()},
+                    "fastqfail": (folders.get("fastqfail") as DataFolder).getTheChildren().find {(it as BarcodedFolder).getSampleId()}
+            ]
+            result[sampleId] = map
+        }
+        return result
+    }
+
+    private Map<String, Map<String, DataFolder>> prepareRawData(String sampleId) {
+        final def result = new HashMap()
+        final def folders = [
+                "fast5fail": (folders.get("fast5failed") as DataFolder),
+                "fast5pass": (folders.get("fast5passed") as DataFolder),
+                "fastqpass": (folders.get("fastqpass") as DataFolder),
+                "fastqfail": (folders.get("fastqfail") as DataFolder)
+        ]
+        result.put(sampleId, folders)
+        return result
+    }
+
+    /**
+     * This method returns a list of all log files from the Oxford Nanopore measurement.
+     *
+     * @return List A list of log files from the experiment
+     */
+    List<DataFile> getLogFiles() {
+        return logFiles.collect { it }
+    }
```

Comment:
Publicly available method. When used in ETL think about defining an interface where this is used. Is there one already in place?

Reply:
No, and as this is a simple getter, an interface is not necessary :)